ssl/quic/quic_impl.c

```diff
@@ -2509,12 +2509,13 @@ static int quic_validate_for_write(QUIC_XSO *xso, int *err)
         /* FALLTHROUGH */
     case QUIC_SSTREAM_STATE_SEND:
     case QUIC_SSTREAM_STATE_DATA_SENT:
-    case QUIC_SSTREAM_STATE_DATA_RECVD:
         if (ossl_quic_sstream_get_final_size(xso->stream->sstream, NULL)) {
             *err = SSL_R_STREAM_FINISHED;
             return 0;
         }
-
+        return 1;
+    case QUIC_SSTREAM_STATE_DATA_RECVD:
+        *err = SSL_R_STREAM_FINISHED;
         return 1;
```

Comment:
I wonder if `return 1` here is correct. We used to return 0 when setting `*err` to `SSL_R_STREAM_FINISHED` This is the only concern I have here.

Reply:
Hmm.. right. It is wrong.

Reply:
fixed